src/common/MPPProber.cc

```diff
@@ -0,0 +1,138 @@
+#include <kvproto/mpp.pb.h>
+#include <pingcap/Exception.h>
+#include <pingcap/common/MPPProber.h>
+#include <pingcap/kv/Cluster.h>
+
+#include <thread>
+
+namespace pingcap
+{
+namespace common
+{
+
+bool MPPProber::isRecovery(const std::string & store_addr, const std::chrono::seconds & recovery_ttl)
+{
+    FailedStoreMap copy_failed_stores;
+    {
+        std::lock_guard<std::mutex> lock(store_lock);
+        copy_failed_stores = failed_stores;
+    }
+    auto iter = copy_failed_stores.find(store_addr);
+    if (iter == copy_failed_stores.end())
+        return true;
+
+    {
+        std::lock_guard<std::mutex> lock(iter->second->state_lock);
+        return iter->second->recovery_timepoint != INVALID_TIME_POINT && getElapsed(iter->second->recovery_timepoint) > recovery_ttl;
```

Comment:
why here need `getElapsed(iter->second->recovery_timepoint) > recovery_ttl`?

Reply:
Max recovery time is `15mins`, recovery_ttl is 60s. See logic in tidb code.

Reply:
Aka, request will succeed after `60secs` if send to this recoveried store explicitly.
background scan will remove this failed store after `15mins`.